tools/test_api.py

```diff
@@ -2201,12 +2201,12 @@ def build_tests(tests, base_source_paths, build_path, target, toolchain_name,
                             report[target_name][toolchain_name][test_key] = report_entry[test_key]
                         
                         # Set the overall result to a failure if a build failure occurred
-                        if not worker_result['result'] and not isinstance(worker_result['reason'], NotSupportedException):
+                        if 'reason' in worker_result and not isinstance(worker_result['reason'], NotSupportedException):
```

Comment:
For this line I think you still need to have the `not worker_result['result']` clause as well. So:

```
if 'reason' in worker_result and not worker_result['result'] and not isinstance(worker_result['reason'], NotSupportedException):
```